python/fledge/services/core/api/plugins/config_validator.py

```diff
@@ -253,86 +348,147 @@ async def ping_host(self, hostname):
+                    return True, f"Host '{hostname}' is reachable but no common services are running on tested ports"
+                elif timeout_count == total_attempts:
+                    # All connections timed out - host is likely down or blocking connections
+                    return False, f"Host '{hostname}' appears to be down or blocking connections (all connection attempts timed out)"
+                elif connection_refused_count > 0:
+                    # Some connections refused - host is likely up
+                    return True, f"Host '{hostname}' is reachable (host responded with connection refused, indicating it's online)"
+                else:
+                    # No clear pattern - host is likely unreachable
+                    return False, f"Host '{hostname}' appears to be unreachable - no successful connections established"
                 
             except socket.gaierror as e:
                 error_msg = str(e).lower()
-                _logger.error(f"DNS resolution failed for {hostname}: {e}")
+                _logger.error(f"DNS resolution failed for the hostname '{hostname}': {e}")
                 
-                if 'name or service not known' in error_msg or 'nodename nor servname provided' in error_msg:
+                if 'name or service not known' in error_msg or 'nodename nor service name provided' in error_msg:
                     return False, f"Cannot resolve hostname '{hostname}' - please check the hostname is correct"
                 elif 'temporary failure' in error_msg:
                     return False, f"Temporary DNS failure for '{hostname}' - please try again later"
                 else:
                     return False, f"DNS lookup failed for '{hostname}'"
-        except asyncio.TimeoutError:
-            _logger.warning(f"Host reachability test timed out for {hostname}")
-            return False, f"Connection test to '{hostname}' timed out - host may be unreachable"
+                    
+        except Exception as e:
+            _logger.error(f"Unexpected error during TCP connectivity test for the hostname '{hostname}': {e}")
+            return False, f"Cannot test connectivity to the hostname '{hostname}' - network error occurred"
+
+    async def ping_host(self, hostname):
+        """
+        Perform host reachability test using ICMP ping when available,
+        falling back to TCP connectivity testing in restricted environments.
+        
+        This method tries ICMP ping first (most reliable and appropriate for host reachability),
+        and only falls back to TCP socket testing when ICMP is not available due to
+        container restrictions or permissions.
+        
+        Args:
+            hostname (str): Hostname or IP address to test
+            
+        Returns:
+            tuple: (success, reason)
+        """
+        try:
+            # Detect if we're in a container environment
+            in_container = self._detect_container_environment()
```

Comment:
You don't really need to do this. Presumably ping will fail in container environments, not found or permissions error. You can se that to fallback to trying to connect to known ports. Testing for containers environments is well known for its issues with all the different variants.

Reply:
removed _detect_container_environment() method